peachpy/x86_64/operand.py

```diff
@@ -427,11 +427,16 @@ def __add__(self, extra_offset):
     def __sub__(self, extra_offset):
         return RIPRelativeOffset(self.offset - extra_offset)
 
+    def __str__(self):
+        return self.format("nasm")
+
     def format(self, assembly_format):
-        if self.offset > 0:
-            return "+%d" % self.offset
+        if assembly_format == "gas":
+            return "%d(%%rip)" % self.offset
+        elif assembly_format == "go":
+            return "%d(IP)" % self.offset
         else:
-            return "%d" % self.offset
+            return "rip%+d" % self.offset
```

Comment:
Typo: should be `"rip+%d"`

Reply:
Actually, it's on purpose --- the `"rip+%d"` will not work if offset is negative, as it will print `rip+-3` instead of `rip-3`. The `%+d` enforces to print either `+3` or `-3`.

Reply:
Ok, LGTM